mongo/main.tf

```diff
@@ -0,0 +1,62 @@
+resource "mongodbatlas_project" "main" {
+  name   = var.atlas_project_name
+  org_id = var.atlas_org_id
+}
+
+resource "mongodbatlas_cluster" "cluster" {
+  project_id = mongodbatlas_project.main.id
+  name       = var.cluster_name
+
+  # Provider Settings "block"
+  provider_name = "TENANT"
+
+  # options: AWS AZURE GCP
+  backing_provider_name = "AWS"
+
+  # options: M2/M5 atlas regions per cloud provider
+  # GCP - CENTRAL_US SOUTH_AMERICA_EAST_1 WESTERN_EUROPE EASTERN_ASIA_PACIFIC NORTHEASTERN_ASIA_PACIFIC ASIA_SOUTH_1
+  # AZURE - US_EAST_2 US_WEST CANADA_CENTRAL EUROPE_NORTH
+  # AWS - US_EAST_1 US_WEST_2 EU_WEST_1 EU_CENTRAL_1 AP_SOUTH_1 AP_SOUTHEAST_1 AP_SOUTHEAST_2
+  provider_region_name = upper(var.region)
+
+  # options: M0 M2 M5
+  provider_instance_size_name = var.cluster_instance_size
+
+  # Will not change till new version of MongoDB but must be included
+  mongo_db_major_version       = "5.0"
+  auto_scaling_disk_gb_enabled = "false"
+
+  lifecycle {
+    prevent_destroy = true
+  }
```

Comment:
Should this be a variable, defaulting to true?  We might use this for throwaway dev db's...

Reply:
Sure - was just thinking about that too - but agreed - good point.  It's a pain to turn that off for dev envs.

Reply:
Ah this can't be a variable - we'd need to create a copy of the cluster and use a different value...

Reply:
We could remove it, have what is basically a separate copy of the module since you'd need a different resource for all of them, or keep it and deal with it if we need to remove the cluster.

Reply:
https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#literal-values-only

Reply:
I would honestly almost just make a dev cluster module - that locks you into M0, maybe a few other settings

Reply:
There's a part of me wondering if there's enough value in putting this in a module, or if we just want a template to copy-paste.